ui/components/app/snaps/snap-ui-address-input/snap-ui-address-input.tsx

```diff
@@ -0,0 +1,189 @@
+import React, {
+  ChangeEvent,
+  FunctionComponent,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { useSelector } from 'react-redux';
+import classnames from 'classnames';
+import { debounce } from 'lodash';
+import {
+  CaipAccountId,
+  CaipChainId,
+  parseCaipAccountId,
+} from '@metamask/utils';
+import {
+  Box,
+  FormTextField,
+  FormTextFieldProps,
+  FormTextFieldSize,
+  Icon,
+  IconName,
+  Label,
+  Text,
+} from '../../../component-library';
+import { useSnapInterfaceContext } from '../../../../contexts/snaps';
+import { getAccountInfoByCaipChainId } from '../../../../selectors/selectors';
+import {
+  AlignItems,
+  BackgroundColor,
+  BorderColor,
+  BorderRadius,
+  Display,
+  FlexDirection,
+  FontWeight,
+  IconColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import Jazzicon from '../../../ui/jazzicon';
+
+export type SnapUIAddressInputProps = {
+  name: string;
+  form?: string;
+  label?: string;
+  chainId: CaipChainId;
+};
+
+export const SnapUIAddressInput: FunctionComponent<
+  SnapUIAddressInputProps & FormTextFieldProps<'div'>
+> = ({ name, form, label, chainId, error, ...props }) => {
+  const { handleInputChange, getValue, focusedInput, setCurrentFocusedInput } =
+    useSnapInterfaceContext();
+
+  const inputRef = useRef<HTMLDivElement>(null);
+
+  const initialValue = getValue(name, form) as string;
+
+  const [value, setValue] = useState(
+    initialValue
+      ? parseCaipAccountId(initialValue as CaipAccountId).address
+      : '',
+  );
+  const [matchedAddressName, setMatchedAddressName] = useState<string | null>(
+    null,
+  );
+
+  const accountsInfo: Record<string, string> = useSelector((state) =>
+    getAccountInfoByCaipChainId(state, chainId),
+  );
+
+  useEffect(() => {
+    if (initialValue !== undefined && initialValue !== null) {
+      const { address } = parseCaipAccountId(initialValue as CaipAccountId);
+      setValue(address);
+    }
+  }, [initialValue]);
```

Comment:
Why is this necessary? Isn't it already being set as initial value?

Reply:
Removed